avocado/plugins/multiplex.py

```diff
@@ -63,34 +86,37 @@ def run(self, args):
         if err:
             log.error(err)
             sys.exit(exit_codes.AVOCADO_FAIL)
-        variants = args.avocado_variants
+        varianter = args.avocado_variants
         try:
-            variants.parse(args)
+            varianter.parse(args)
         except (IOError, ValueError) as details:
-            log.error("Unable to parse variants: %s", details)
+            log.error("Unable to parse varianter: %s", details)
             sys.exit(exit_codes.AVOCADO_JOB_FAIL)
+        use_utf8 = settings.get_value("runner.output", "utf8",
+                                      key_type=bool, default=None)
+        summary = args.summary or 0
+        variants = args.variants or 0
+
+        # Parse obsolete options (unsafe to combine them with new args)
         if args.tree:
+            summary += 1
             if args.contents:
-                verbose = 1
-            else:
-                verbose = 0
+                summary += 1
             if args.inherit:
-                verbose += 2
-            use_utf8 = settings.get_value("runner.output", "utf8",
-                                          key_type=bool, default=None)
-            log.debug(tree.tree_view(variants.variants.root, verbose, use_utf8))
-            sys.exit(exit_codes.AVOCADO_ALL_OK)
-
-        log.info('Variants generated:')
-        if args.mux_debug:
-            # In this version `avocado_variants.debug` is not set properly,
-            # let's force-enable it before calling to_str to
-            # get the expected results.
-            args.avocado_variants.debug = True
-        use_utf8 = settings.get_value("runner.output", "utf8",
-                                      key_type=bool, default=None)
-        variants = args.avocado_variants.to_str(0, 3 if args.contents else 2)
-        for line in variants.splitlines():
+                summary += 2
+        else:
+            if args.contents:
+                variants += 2
+
+        # When nothing is specified, show variants
+        if not summary and not variants:
+            variants = 1
```

Comment:
I know, I could use `None` if you believe this should be supported. I considered this as a mistake by the user input...